superset/assets/javascripts/explorev2/components/SaveModal.js

```diff
@@ -0,0 +1,245 @@
+/* eslint camel-case: 0 */
+import React, { PropTypes } from 'react';
+const $ = window.$ = require('jquery');
+import { Modal, Alert } from 'react-bootstrap';
+import Select from 'react-select';
+import { connect } from 'react-redux';
+
+const propTypes = {
+  can_edit: PropTypes.bool,
+  onHide: PropTypes.func.isRequired,
+  actions: PropTypes.object.isRequired,
+  form_data: PropTypes.object,
+  datasource_type: PropTypes.string.isRequired,
+  user_id: PropTypes.string.isRequired,
+};
+
+class SaveModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      save_to_dashboard_id: null,
+      dashboards: [],
+      alert: null,
+    };
+  }
+
+  componentDidMount() {
+    this.prepSaveDialog();
+  }
+
+  setSaveDashboard(opt) {
+    this.setState({ save_to_dashboard_id: opt.value });
+    $('#add_to_dash_existing').prop('checked', true);
+    $('.gotodash').removeAttr('disabled');
+  }
+
+  fetchDashboards() {
+    const url = '/dashboardmodelviewasync/api/read?_flt_0_owners=' + this.props.user_id;
+    $.get(url, function (data) {
```

Comment:
we may not use this action in other components yet, but it's great to keep all the fetching logic in one place and keep the components leaner. could be an opportunity to start an actions file for dashboard actions, but i think we could put this fetch in the explore actions for now.